From david-b@pacbell.net Mon Apr 23 10:50:22 2007
From: Erik Hovland <erik@hovland.org>
Date: Mon, 23 Apr 2007 10:50:15 -0700
Subject: usb ethernet gadget, workaround network stack API glitch
To: Greg KH <greg@kroah.com>
Cc: Erik Hovland <erik@hovland.org>
Message-ID: <200704231050.15344.david-b@pacbell.net>
Content-Disposition: inline


From: Erik Hovland <erik@hovland.org>

Another workaround for the glitch in the network layer, whereby one call
ignores the (otherwise kernel-wide) convention that free() calls should
not oops when passed nulls.  This code already handles that API glitch in
most other paths.

From: Erik Hovland <erik@hovland.org>
Signed-off-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/gadget/ether.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/usb/gadget/ether.c
+++ b/drivers/usb/gadget/ether.c
@@ -1735,7 +1735,8 @@ enomem:
 		defer_kevent (dev, WORK_RX_MEMORY);
 	if (retval) {
 		DEBUG (dev, "rx submit --> %d\n", retval);
-		dev_kfree_skb_any (skb);
+		if (skb)
+			dev_kfree_skb_any(skb);
 		spin_lock(&dev->req_lock);
 		list_add (&req->list, &dev->rx_reqs);
 		spin_unlock(&dev->req_lock);
